src/main/java/io/gravitee/fetcher/github/GitHubFetcher.java

```diff
@@ -325,6 +325,18 @@ public void handle(HttpClientRequest request) {
                                             // Close client
                                             httpClient.close();
                                         });
+                                    } else if (response.statusCode() == HttpStatusCode.NOT_FOUND_404) {
+                                        promise.fail(
+                                            new ResourceNotFoundException(
+                                                "Unable to fetch '" +
+                                                url +
+                                                "'. Status code: " +
+                                                response.statusCode() +
+                                                ". Message: " +
+                                                response.statusMessage(),
+                                                null
```

Comment:
maybe we can just go with the `unable to fetch ${URL}` part of the message WDYT ?